src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpListener.java

```diff
@@ -11,29 +13,41 @@
 import net.dv8tion.jda.api.hooks.ListenerAdapter;
 import net.dv8tion.jda.api.interactions.components.ActionComponent;
 import net.dv8tion.jda.api.interactions.components.buttons.Button;
-import net.javadiscord.javabot.Bot;
+import net.javadiscord.javabot.data.config.BotConfig;
 import net.javadiscord.javabot.data.config.GuildConfig;
 import net.javadiscord.javabot.data.config.guild.HelpConfig;
+import net.javadiscord.javabot.data.h2db.DbActions;
 import net.javadiscord.javabot.systems.help.HelpChannelManager;
 import net.javadiscord.javabot.systems.help.HelpExperienceService;
+import net.javadiscord.javabot.systems.help.dao.HelpAccountRepository;
+import net.javadiscord.javabot.systems.help.dao.HelpTransactionRepository;
 import net.javadiscord.javabot.systems.help.model.HelpTransactionMessage;
 import net.javadiscord.javabot.util.ExceptionLogger;
 import net.javadiscord.javabot.util.Responses;
 import org.jetbrains.annotations.NotNull;
+import org.springframework.dao.DataAccessException;
 
-import java.sql.SQLException;
 import java.util.*;
 
+import javax.sql.DataSource;
+
 /**
  * Listens for all events releated to the forum help channel system.
  */
+@RequiredArgsConstructor
 public class ForumHelpListener extends ListenerAdapter implements ButtonHandler {
```

Comment:
Subtypes of `ListenerAdapter` are automatically detected as per the filters in `Bot`.

Reply:
Yeah, but this class is using a ButtonHandler

Reply:
As soon as it implements `ListenerAdapter`, it is managed by Spring. Since it is managed by Spring, it is detecting when asking spring for instances of `ButtonHandler`.

Reply:
My fault, I fixed that.